### ISSUE TITLE
Experiment switching to GitHub Actions for CI

### ISSUE THREAD

**Issue by kitchoi** [enthought/traitsui#1387]
Motivation: Currently the per-PR CI build is quite slow, especially after switching the OSX builds over from Travis to Appveyor. By the time Travis is done completing four jobs, Appveyor is often still running its first or second job. This slowness actually motivates PRs to lump independent bits and pieces together just to save CI time.

The main bottleneck here, I believe, is that only one concurrent job is allowed to run at a time on Appveyor.

This issue proposes experimenting with switching CI to GitHub Actions, which, as of writing, allows multiple jobs to run in parallel. I believe there are other benefits (and maybe costs), but the main motivation here is for shortening per-PR CI build time. Shall this issue be closed and the CI build is still slow, then we may need another issue purely for the speed issue.

**Comment by mdickinson** [comment]
Related Traits issue, that summarizes some of the tradeoffs of GitHub Actions: https://github.com/enthought/traits/pull/1287

**Comment by mdickinson** [comment]
A random note: core Python recently decided to drop Travis CI as a requirement because of its flakiness. See [this thread](https://mail.python.org/archives/list/python-committers@python.org/thread/JR7HIQHVECHPKH2LE46EACN73KCBLVKE/) for more.

**Comment by kitchoi** [comment]
From #1415, it is a decent speed up: All the jobs for GitHub Actions finished within about 5 minutes. Previously, we would have to be waiting for Appveyor to complete after ~30 minutes.
#1415 will close this issue for the purpose of experimentation. I will open a separate issue to track all the jobs we need to migrate.
